decidim-core/app/commands/decidim/create_amend.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Decidim
+  # A command with all the business logic when a user starts amending a resource.
+  class CreateAmend < Rectify::Command
+    # Public: Initializes the command.
+    #
+    # form         - A form object with the params.
+    # amendable    - The resource that is being amended.
+    def initialize(form)
+      @form = form
+      @amendable = form.amendable
+    end
+
+    # Executes the command. Broadcasts these events:
+    #
+    # - :ok when everything is valid, together with the amend.
+    # - :invalid if the form wasn't valid and we couldn't proceed.
+    #
+    # Returns nothing.
+    def call
+      return broadcast(:invalid) if form.invalid?
+
+      transaction do
+        create_emendation!
+        create_amend!
+
+        # The proposal authors and followers are notified that an amendment has been created.
+        notify_amendable_authors_and_followers
+      end
+
+      broadcast(:ok)
+    end
+
+    private
+
+    attr_reader :amendment, :form, :current_user
+
+    def create_emendation!
+      @emendation = Decidim.traceability.create!(
+        form.amendable_type.constantize,
+        form.current_user,
+        emendation_attributes
+      )
+      @emendation.add_coauthor(form.current_user, user_group: nil)
+    end
+
+    def emendation_attributes
+      {
+        title: "[emendation] #{form.title}",
+        body: form.body,
+        component: form.amendable.component,
+        published_at: Time.current
```

Comment:
Can emendations not be published?

Reply:
well, this needs more work to abstract it to a general case.
In this case as the emmendations are a `Proposal` requires a `published_at`